CONTRIBUTING.md

```diff
@@ -1,143 +1,266 @@
+essentially submodules within the repository. These are centered around:
 
-1. Save (rebuild) any patches you are in the middle of working on!
-1. Identify the names of the files you want to import.
-   - A complete list of all possible file names can be found at ```./work/Minecraft/$MCVER/spigot/net/minecraft/server```
-1. Open the file at `./scripts/importmcdev.sh` and add the name of your file to the script.
-1. Re-patch the server `./paper patch`
-1. Edit away!
+- `Paper-API` (the `Spigot-API`/`Bukkit` for Paper);
+- `Paper-MojangAPI` (API for
+[Mojang's Brigadier](https://github.com/Mojang/brigadier));
+- `Paper-Server` (the `Spigot`/`CraftBukkit` for Paper).
 
-This change is temporary! DO NOT COMMIT CHANGES TO THIS FILE!
-Once you have made your changes to the new file, and rebuilt patches, you may undo your changes to importmcdev.sh
+Because the entire structure is based on patches and git, a basic understanding
+of how to use git is required. A basic tutorial can be found here:
+<https://git-scm.com/docs/gittutorial>.
 
-Any file modified in a patch file gets automatically imported, so you only need this temporarily
-to import it to create the first patch.
+Assuming you already have forked the repository:
+
+1. Pull the latest changes from the main repository;
+1. Type `./paper patch` in git bash to apply the changes from upstream;
+1. cd into `Paper-Server` for server changes, and `Paper-API` for API changes.  
+You can also run `./paper server` or `./paper api` for these same directories
+respectively. If you install the Paper tool with `./paper setup`, you can run
+`paper <command>` from anywhere in most cases.
+
+These directories aren't git repositories in the traditional sense:
+
+- Every single commit in `Paper-Server`/`Paper-API` is a patch;
+- `origin/master` points to a directory similar to `Paper-Server`/`Paper-API`
+but for Paper;
+- Typing `git status` should show that we are 10 or 11 commits ahead of master,
+meaning we have 10 or 11 patches Spigot and CraftBukkit don't.
+   - If it says something like `212 commits ahead, 207 commits behind`,
+   cd into the root directory of the cloned repository and type `git fetch` to
+   update your upstream. Setting up a remote for the upstream Paper repository
+   might be necessary.
+
+## Adding Patches
+
+Adding patches to Paper is very simple:
 
-To undo your changes to the file, type `git checkout scripts/importmcdev.sh`
+1. Modify `Paper-Server` and/or `Paper-API` with the appropriate changes;
+1. Type `git add .` inside these directories to add your changes;
+1. Run `git commit` with the desired patch message;
+1. Run `./paper rebuild` in the main directory to convert your commit into a new
+patch;
+1. PR your patches back to this repository.
 
+Your commit will be converted into a patch that you can then PR into Paper.
+
+❗ Please note that if you have some specific implementation detail you'd like
+to document, you should do so in the patch message *or* in comments.
```

Comment:
```suggestion
> ❗ Please note that if you have some specific implementation detail you'd like
to document, you should do so in the patch message *or* in comments.
```
Placing noteworthy messages in blockquotes will make them more noticeable than just an emoji at the start of the sentence

Reply:
The block quotes are (sadly) not rendered in GitHub, but I've changed to use them for anyone viewing the document in something like IntelliJ's markdown previewer.